task/task.go

```diff
@@ -0,0 +1,42 @@
+package task
+
+import (
+	"errors"
+	"log"
+
+	"github.com/symflower/eval-dev-quality/language"
+)
+
+var (
+	// ErrTaskUnsupported indicates that a task is unsupported.
+	ErrTaskUnsupported = errors.New("task unsupported")
+)
+
+// Identifier holds the identifier of a task.
+type Identifier string
```

Comment:
I like this to be reworded. In evaluate/metrics/assessment.go we have "Key". @bauersimon @ahumenberger any other suggestions? "Identifier" doesn't sound right to me.

Reply:
Each Model, Provider, Tool on the other hand does have an `ID`. But that is a slightly different structure.

Reply:
Or since it is the only thing that currently identifies what a task is, why not call it `Task`?

Reply:
And then `AllIdentifiers` -> `AllTasks`.

Reply:
I see. That brings me back to my original request i had with Andi: making the tasks use methods of models/languages instead of letting the models reimplement tasks. Then every task has the same pattern as model/language/provider/...

Reply:
The assessment type is basically a database, so "Key" fits perfectly there, but it does not for tasks. A task identifier identifies a task. E.g. a repository can  list tasks which are supposed to be evaluated. Sounds right to me.

Reply:
> making the tasks use methods of models/languages instead of letting the models reimplement tasks.

This is meant to be done as a follow-up. This PR only introduces the primitive concept running different tasks. For this, a "task identifier" is sufficient. Later a `Task` type can be introduced. A "task identifier" is still necessary, because e.g. a repository can specify a list of tasks that should be run.

Reply:
Then let's do `Key` for now? And when we touch it again, we can see what makes most sense then.

Reply:
Since I think "Identifier" is totally fine, and there are no better suggestions, I'm going to leave it.